nautobot/extras/tests/test_customfields.py

```diff
@@ -1457,105 +1459,433 @@ def setUpTestData(cls):
         Site.objects.create(name="Site 4", slug="site-4", _custom_field_data={})
 
     def test_filter_integer(self):
-        self.assertEqual(self.filterset({"cf_cf1": 100}, self.queryset).qs.count(), 1)
-        self.assertEqual(self.filterset({"cf_cf1__n": [100]}, self.queryset).qs.count(), 3)
-        self.assertEqual(self.filterset({"cf_cf1__lte": [101]}, self.queryset).qs.count(), 1)
-        self.assertEqual(self.filterset({"cf_cf1__lt": [101]}, self.queryset).qs.count(), 1)
-        self.assertEqual(self.filterset({"cf_cf1__gte": [199]}, self.queryset).qs.count(), 1)
-        self.assertEqual(self.filterset({"cf_cf1__gt": [199]}, self.queryset).qs.count(), 1)
+        self.assertQuerysetEqual(
+            self.filterset({"cf_cf1": 100}, self.queryset).qs,
+            self.queryset.filter(_custom_field_data__cf1=100),
+        )
+        self.assertQuerysetEqual(
+            self.filterset({"cf_cf1__n": [100]}, self.queryset)
+            .qs.exclude(_custom_field_data={})
+            .exclude(_custom_field_data__cf1__isnull=True),
+            self.queryset.exclude(_custom_field_data__cf1=100),
```

Comment:
This approach throughout this file *feels* wrong to me. We shouldn't be modifying what the `self.filterset` evaluates to as part of the test - if we need to do further manipulation of `self.queryset` to get the same result as the filterset, let's do that instead.